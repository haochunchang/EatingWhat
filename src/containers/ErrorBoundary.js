import React, { Component } from "react";
import "../components/App.css";

class ErrorBoundary extends Component {
  constructor() {
    super();
    this.state = {
      hasError: false,
      errormsg: "",
      infomsg: "",
    };
  }

  componentDidCatch(error, info) {
    this.setState({
      hasError: true,
      errormsg: error,
      infomsg: info,
    });
  }

  render() {
    const { hasError, errormsg, infomsg } = this.state;
    if (hasError) {
      return (
        <div>
          <h1>Oops, somthing wrong happens...</h1>
          <p>The error message is: {errormsg}.</p>
          <p>The info message is {infomsg}.</p>
        </div>
      );
    }
    return this.props.children;
  }
}

export default ErrorBoundary;
