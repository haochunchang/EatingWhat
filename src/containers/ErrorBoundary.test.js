import React from "react";
import { shallow } from "enzyme";
import ErrorBoundary from "./ErrorBoundary";

describe("ErrorBoundary", () => {
  const wrapper = shallow(<ErrorBoundary />);
  const componentInstance = wrapper.instance();

  it("shallow rendering correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("initial states of error boundary", () => {
    expect(componentInstance.state.hasError).toBeFalsy();
    expect(componentInstance.state.errormsg).toBeFalsy();
    expect(componentInstance.state.infomsg).toBeFalsy();
  });

  it("component did catch error", () => {
    const error = "Testing error!!";
    const info = "This is the info!";
    componentInstance.componentDidCatch(error, info);
    expect(componentInstance.state.hasError).toBeTruthy();
    expect(componentInstance.state.errormsg).toEqual(error);
    expect(componentInstance.state.infomsg).toEqual(info);
  });
});
