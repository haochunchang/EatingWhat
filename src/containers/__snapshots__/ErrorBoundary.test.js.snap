// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary shallow rendering correctly 1`] = `ShallowWrapper {}`;
